### PR TITLE
Resolved issue #61 - Ignore the non-content elements in the search output

### LINKED ISSUE #61: Ignore the non-content elements in the search output
**Describe the bug**
The search function currently includes non-content elements (such as titles, dotted lines, and headers) in the search output. These elements are counted towards the specified number of lines. The function should ignore these elements and only include the actual content lines in the search results.

**To Reproduce**
1. Run the application and navigate to https://localhost:3000
2. Choose the ORCA log file as the input and upload it.
3. Provide the following search inputs:
-   Search term (e.g., "CARTESIAN COORDINATES (A.U.)")
- Lines specified (e.g., FIRST 5)
- Number of sections (e.g., 1, 2)
5. Click on Preview Output
6. Observe that the title, dotted line, and header are incorrectly included in the output

**Expected behavior**
Using the search criteria above, the output should display only the first 5 lines of actual content for "CARTESIAN COORDINATES (A.U.)" from cycles 1 and 2. Empty lines, titles, dotted lines, and headers should be ignored in the search results.

**Current behavior**
The search function incorrectly considers non-content elements (such as titles and formatting lines) as part of the output. This results in displaying fewer lines of relevant content, as these elements are counted towards the specified number of lines. For example, with the current implementation, only 2 lines of actual content are shown instead of the expected 5 lines.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Lyskawa-Aonyx**
Fixes #61 - Ignore the non-content elements in the search output

**What was changed?**
I added a few lines of code in the app.py file to handle this issue
*Here, describe what part of the application you changed (e.g. login page, database, etc.). If possible, mention what specific components were added, removed, or modified.*
backend text processing 
**Why was it changed?**
to ensure that non-relevant chars would not be included in the output. 
*Here, describe the issue that you are fixing with this code, and why your code fixes it.*
I changed parts of the extract_sections function in app.py to handle this issue correctly and to only show relevant chars in the output preview. 
**How was it changed?**
changed lines of code
*Here, get into detail about what files you modified, and talk about the most important lines in regards to fixing the issue.*

**Screenshots that show the changes (if applicable):**

### CODE REVIEW

**Review by mkalal6 — COMMENTED** [review]
Is there a lot of repetition of is_content_line that can be reduced? Otherwise good code.

File: server/services/file_search_operations.py [hunk 1]
```diff
@@ -25,41 +24,49 @@ def extract_sections(file_path, search_terms, sections, specify_lines, use_total
         start_line = term_line_num[i-1]
         line_empty = 0
```
Comment (SiriChandanaGarimella) [inline server/services/file_search_operations.py:25]: Please add the search_term as follows:
 search_term = search_terms[0] # for now (we will need to make it generic later) 
 and pass it as the parameter value to the is_content_line() function.


Also, add the following as the header_pattern above the for-loop (line 22) and pass it as the parameter value to the is_content_line() function:
header_pattern = r'^\s*NO\s+LB\s+ZA\s+FRAG\s+MASS\s+X\s+Y\s+Z\s*$'

**Review by SiriChandanaGarimella — COMMENTED** [review]
@Lyskawa-Aonyx - The changes look good, and you are in the right direction. Please check the following points:
1. Consider the search term as the title since there is no text labeled 'Title'.
2. Add a regex to ignore the header of the search term: CARTESIAN COORDINATES (A.U.)
I have added inline comments. Please check.

Also, for the next issue, please use the issue number as the branch name (e.g., issue_61).

**Review by SiriChandanaGarimella — CHANGES_REQUESTED** [review]
Please check the added inline comments

**Pull request closed without merging**